src/Features/CSharp/Portable/DocumentationComments/DocumentationCommentSnippetService.cs

```diff
@@ -111,22 +124,106 @@ protected override List<string> GetDocumentationCommentStubLines(MemberDeclarati
+        {
+            var throwExpressionsAndStatements = member.DescendantNodes().Where(n => n.IsKind(SyntaxKind.ThrowExpression, SyntaxKind.ThrowStatement));
+            var namespacesInScope = model.GetUsingNamespacesInScope(member);
+            var hasUsingSystem = namespacesInScope.Contains(n => n.ContainingNamespace.IsGlobalNamespace && n.Name == "System");
+            using var _ = PooledHashSet<string>.GetInstance(out var seenExceptionTypes);
+
+            foreach (var throwExpressionOrStatement in throwExpressionsAndStatements)
+            {
+                var expression = throwExpressionOrStatement switch
+                {
+                    ThrowExpressionSyntax throwExpression => throwExpression.Expression,
+                    ThrowStatementSyntax throwStatement => throwStatement.Expression,
+                    _ => throw ExceptionUtilities.Unreachable
+                };
+
+                if (expression.IsKind(SyntaxKind.NullLiteralExpression))
+                {
+                    // `throw null;` throws NullReferenceException at runtime.
+                    var exception = hasUsingSystem ? nameof(NullReferenceException) : $"{nameof(System)}.{nameof(NullReferenceException)}";
+                    if (seenExceptionTypes.Add(exception))
+                    {
+                        yield return exception;
+                    }
+                }
+                else if (expression is not null)
+                {
+                    var type = model.GetTypeInfo(expression, cancellationToken).Type;
+                    if (type is not null && !IsExceptionCaughtAndNotRethrown(throwExpressionOrStatement, type, model, cancellationToken))
+                    {
+                        var exception = type.ToDisplayString();
+                        if (seenExceptionTypes.Add(exception))
+                        {
+                            if (namespacesInScope.Contains(n => n.ToDisplayString() == type.ContainingNamespace.ToString()))
+                            {
+                                yield return type.ToDisplayString(s_format).Replace('<', '{').Replace('>', '}');
+                            }
+                            else
+                            {
+                                yield return exception.Replace('<', '{').Replace('>', '}');
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static bool IsExceptionCaughtAndNotRethrown(SyntaxNode throwExpressionOrStatement, ITypeSymbol exceptionType, SemanticModel model, CancellationToken cancellationToken)
+        {
+            var tryStatement = throwExpressionOrStatement.FirstAncestorOrSelfUntil<TryStatementSyntax>(n => n is CatchClauseSyntax or LocalFunctionStatementSyntax or MemberDeclarationSyntax);
```

Comment:
i think you'd need to walk up *all* containing try statements, not jsut the immediate one.